ingestors/kafka/src/main/scala/hydra/kafka/algebras/KafkaClientAlgebra.scala

```diff
@@ -304,24 +326,42 @@ object KafkaClientAlgebra {
         consumerStream(consumerSettings)
           .evalTap(_.subscribeTo(topicName))
           .flatMap(_.stream)
-          .evalTap{ committable =>
+          // TODO: Commit before action? do we need at least once delivery or at most once?
+          .evalTap { committable =>
             if (commitOffsets) committable.offset.commit else Applicative[F].pure(())
           }
           .map { committable =>
             val r = committable.record
             val headers = if (r.headers.isEmpty) None else Option(r.headers)
-            (r.key, r.value, headers, r.partition, r.offset)
-          }
+
+            for {
+              key <- eitherToOption(r.key, deserializationErrorMessage(r, topicName))
+              value <- eitherToOption(r.value, deserializationErrorMessage(r, topicName))
+            } yield (key, value, headers, r.partition, r.offset)
+          }.collect {
+          case Some(value) => value
+        }
       }
 
+      private def eitherToOption[T](either: Either[Throwable, T], errorMessage: String): Option[T] = either match {
+        case Right(value) => Some(value)
+        case Left(error) =>
+          logger.warn(errorMessage, error)
```

Comment:
This will go only to our main file system. To go to KStreams jobs we need to fix this issue separately because they are using different deserializators and consuming mechanism